miner/worker.go

```diff
@@ -915,6 +958,21 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 			log.Trace("Skipping unsupported transaction type", "sender", from, "type", tx.Type())
 			txs.Pop()
 
+		case errors.Is(err, circuitcapacitychecker.ErrBlockRowConsumptionOverflow):
+			// Circuit capacity limit reached in a block, don't pop or shift, just quit the loop immediately
+			log.Trace("Circuit capacity limit reached in a block", "acc_rows", w.current.accRows, "tx", tx.Hash())
+			break loop
+
+		case errors.Is(err, circuitcapacitychecker.ErrTxRowConsumptionOverflow):
+			// Tx row consumption too high, discard the tx
+			log.Trace("Circuit capacity limit reached for a single tx", "tx", tx.Hash())
+			txs.Shift()
+
+		case errors.Is(err, circuitcapacitychecker.ErrUnknown):
+			// Unknown circuit capacity checker error, pop, and keep it instead of discard it
+			log.Trace("Unknown circuit capacity checker error", "tx", tx.Hash())
+			txs.Pop()
```

Comment:
We cannot use `Pop` for L1 messages, that way we'll skip messages that we shouldn't skip. Example `txs`:

```
txs = {
  A: [{ QueueIndex = 1 }, { QueueIndex = 2 }],
  B: [{ QueueIndex = 3 }],
}
```

If `1` overflows, we should not skip `2`.

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/411